File: 1. 단계별로_풀어보기/10~19/14. 집합과 맵/4. 1620.py
import math
import sys

input = sys.stdin.readline
output = sys.stdout.write

N, M = map(int, input().split())

pokeArr = list(input().strip() for i in range(N))
queryArr = list(input().strip() for i in range(M))

pokeDict = {}

for i in range(N):
  pokeDict[pokeArr[i]] = i + 1

for query in queryArr:
  if query.isdigit():
    output(f"{pokeArr[int(query) - 1]}\n")
  else:
    output(f"{pokeDict[query]}\n")

File: 1. 단계별로_풀어보기/10~19/14. 집합과 맵/4. 1620.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const [N, M] = inputArr[0].split(' ').map(Number)
const pokeArr = inputArr.slice(1, N + 1).map(row => row.trim())
const queryArr = inputArr.slice(N + 1).map(row => row.trim())

// 순서 저장할 dictionary
const dict = {}

// 순서 저장
pokeArr.forEach((name, idx) => (dict[name] = idx + 1))

queryArr.forEach(query => {
  const idx = +query

  if (idx) process.stdout.write(`${pokeArr[idx - 1]}\n`)
  else process.stdout.write(`${dict[query]}\n`)
})
